pkg/contract/solana/gateway_message.go

```diff
@@ -0,0 +1,73 @@
+package solana
+
+import (
+	"encoding/binary"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/gagliardetto/solana-go"
+)
+
+// MsgWithdraw is the message for the Solana gateway withdraw instruction
+type MsgWithdraw struct {
+	// ChainID is the chain ID of Solana chain
+	ChainID uint64
+
+	// Nonce is the nonce for the withdraw
+	Nonce uint64
+
+	// Amount is the lamports amount for the withdraw
+	Amount uint64
+
+	// To is the recipient address for the withdraw
+	To solana.PublicKey
+
+	// Signature is the signature of the message
+	Signature [65]byte
```

Comment:
Should we add `type Signature [65]byte`?

Reply:
It's okay to use [65]byte for only in single file. The `go-ethereum/crypto/secp256k1` also returns raw bytes on `Sign` and takes raw bytes on `PubkeyToAddress`.